python/tchannel/tornado/tornado_req_handler.py

```diff
@@ -0,0 +1,92 @@
+import json
+from tornado import httputil
+from tornado.httputil import RequestStartLine
+from ..req_handler import RequestHandler
+
+
+class TornadoRequestHandler(RequestHandler):
+    def __init__(self, app):
+        self.request_callback = app
+
+    def start_serving(self, request_conn):
+        return _ServerRequestAdapter(self, request_conn)
+
+    def handle_request(self, context, conn):
+        """Handle incoming request
+
+        :param context: incoming message context
+        :param conn: incoming connection
+        """
+        request_delegate = self.start_serving(conn)
+        message = context.message
+        # process http message
+        if message.headers["as"] == "http":
+            method = "GET"
+            if (hasattr(message, "arg_3") and
+                    message.arg_3 is not None and
+                    message.arg_3 != ""):
+                method = "POST"
+
+            start_line = RequestStartLine(method, message.arg_1, 'HTTP/1.1')
+            try:
+                headers = json.loads(message.arg_2)
+            except:
+                headers = {}
+
+            body = message.arg_3 if hasattr(message, "arg_3") else ""
+            request_delegate.headers_received(start_line, headers)
+            request_delegate.data_received(body)
+            request_delegate.finish()
+
+
+class _ServerRequestAdapter():
+    """Adapts the `TChannelMessageDelegate` interface to the interface expected
```

Comment:
this is a class I move from tornado framework to here. I fake some of the implementation of some of the methods inside this class to work for our tchannel. I won't change any interface or structure of this class. I prefer to keep it same as tornado's.